universal-application-tool-0.0.1/app/services/question/QuestionDefinition.java

```diff
@@ -189,8 +189,22 @@ public ScalarType getProgramIdType() {
     return ScalarType.LONG;
   }
 
-  /** Get a map of scalars stored by this question definition. */
-  public abstract ImmutableMap<Path, ScalarType> getScalars();
+  /** Get a map of all scalars stored by this question definition. */
+  public ImmutableMap<Path, ScalarType> getScalars() {
+    return ImmutableMap.<Path, ScalarType>builder()
+        .putAll(getScalarPaths())
+        .putAll(getMetadataPaths())
+        .build();
+  }
+
+  /** Get a map of question specific scalars stored by this question definition. */
+  abstract ImmutableMap<Path, ScalarType> getScalarPaths();
```

Comment:
Thoughts (non-blocking): Maybe `ScalarMap`? I feel `Path` in the name may be a little bit misleading.

Reply:
changed!